### PR TITLE
tests: Throttle the garbage_collector tests

### LINKED ISSUE #107394: Garbage collector should orphan pods created by rc if delete options say so on Windows causes flakiness
### Which jobs are flaking?

https://testgrid.k8s.io/sig-windows-master-release#capz-windows-containerd-master

### Which tests are flaking?

it is random but I did some analysis here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1932#issuecomment-1006759077

Once I disabled these tests I've seen test success improve.

### Since when has it been flaking?

since this job was enabled

### Testgrid link

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1932/pull-cluster-api-provider-azure-windows-containerd-upstream-with-ci-artifacts/1472004445293252608

### Reason for failure (if possible)

`Garbage collector should orphan pods created by rc if delete options say so` or other similiar GC tests create 30-40+ pods on the VM's. The tests checks that the pods don't get GC'd then delete them with termination-graceperiod=0 and the test completes. This leaves all of those pods in ContainerCreating state until a much later point when the containers are started and then issued an immediate delete. All of this causes strain (high cpu) on the system and other pods end up timing out or other issues. Note this is really only an issue on Windows due to the overhead of starting/deleting containers. These containers can take 10+ mins to clear from the system causing other tests run after to timeout. 

To reproduce:

```
./hack/ginkgo-e2e.sh --node-os-distro=windows --ginkgo.focus="should orphan pods created by rc if delete options"  --ginkgo.skip="\[LinuxOnly\]" --report-dir=/home/jstur/out/kubetest --dump-logs-on-failure=false --disable-log-dump=true --feature-gates="WindowsHostProcessContainers=true"
```

Test passes and completes in 44 seconds:

```
[AfterEach] [sig-api-machinery] Garbage collector
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:186
Jan  6 10:29:44.186: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "gc-2649" for this suite.
•{"msg":"PASSED [sig-api-machinery] Garbage collector should orphan pods created by rc if delete options say so [Conformance]","total":1,"completed":1,"skipped":6374,"failed":0}
```

Watching the containers created during test show they never enter "Running" phase and once test finished they are marked as terminating:

```
kubectl get pods -A -w
gc-2649       simpletest.rc-wmkj4                                            0/1     ContainerCreating   0             30s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-w5mcv                                            0/1     ContainerCreating   0             31s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-gdn6h                                            0/1     ContainerCreating   0             31s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-6k89c                                            0/1     ContainerCreating   0             31s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-tqdj8                                            0/1     ContainerCreating   0             32s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-5l79n                                            0/1     ContainerCreating   0             33s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-6pr5r                                            0/1     ContainerCreating   0             34s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-l4q6c                                            0/1     ContainerCreating   0             34s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-xsznv                                            0/1     ContainerCreating   0             35s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-pwnhl                                            0/1     ContainerCreating   0             36s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-kkpwr                                            0/1     ContainerCreating   0             35s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-gln84                                            0/1     ContainerCreating   0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-x8q6m                                            0/1     ContainerCreating   0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-r2qfz                                            0/1     ContainerCreating   0             36s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-mhvsb                                            0/1     ContainerCreating   0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-c299p                                            0/1     ContainerCreating   0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-2rrft                                            0/1     Terminating         0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-2rrft                                            0/1     Terminating         0             37s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-2s9g6                                            0/1     Terminating         0             38s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-2s9g6                                            0/1     Terminating         0             38s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-7ftwj                                            0/1     ContainerCreating   0             39s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-46bbf                                            0/1     Terminating         0             40s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-46bbf                                            0/1     Terminating         0             40s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-4glbv                                            0/1     Terminating         0             40s                                                                                                                                                                                                                                                          
gc-2649       simpletest.rc-4glbv                                            0/1     Terminating         0             40s
```

Looking at kubelet logs the we can see the containers don't move to "running" till 3+ minutes after the tests completed (18:32:58.693537) later when they are immediately terminated:hcs

```
kubelet.exe.INFO:8584:E0106 18:32:58.693537    4788 event.go:267] Server rejected event                                 '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""},                                                                ObjectMeta:v1.ObjectMeta{Name:"simpletest.rc-c299p.16c7c213994edc14", GenerateName:"", Namespace:"gc-2649",             SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0,      time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil),            Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil),                 ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Pod",              Namespace:"gc-2649", Name:"simpletest.rc-c299p", UID:"274e93ea-713e-4196-9a98-1a3c25a38f36", APIVersion:"v1",           ResourceVersion:"121158", FieldPath:"spec.containers{nginx}"}, Reason:"Created", Message:"Created container nginx",     Source:v1.EventSource{Component:"kubelet", Host:"capz-conf-ftk9c"}, FirstTimestamp:time.Date(2022, time.January, 6,     18, 32, 58, 632084500, time.Local), LastTimestamp:time.Date(2022, time.January, 6, 18, 32, 58, 632084500, time.Local),  Count:1, Type:"Normal", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil),   Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'namespaces "gc-2649"    not found' (will not retry!)                                      
```

and even in some cases failed to start: 

```
kubelet.exe.INFO:8843:E0106 18:34:58.698948    4788 pod_workers.go:918] "Error syncing pod, skipping" err="failed to
\"StartContainer\" for \"nginx\" with RunContainerError: \"context deadline exceeded\""
pod="gc-2649/simpletest.rc-c299p" podUID=274e93ea-713e-4196-9a98-1a3c25a38f36
```

and after the test is completed the containers are in various states on the system:

```
PS C:\Users\capi\ProcessExplorer> hcsdiag list
028a9978ed84e8a4f962e2a5cd0f1e48378751734cb0110e7cd9084036ef3cd2
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

053cba39311ee3612908894c6e1a894f520ab29d2fccadd0a1a6d1c9b561bdcc
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

08ebc13beb5dffe37c86d7f992cf1be6480a32da62e026c9a6d02080ab7eeff0
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

0f8609bd174e489c02b3bf26102be96be7c379c14b07ac298944a18d45269663
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

135bd23192de2310cc1448e1a8e72fe7e9f4ceaf02bf92612eea2eaa61b7192f
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

147dd265fc21ea34437675c48b84d9bda2d3bf4c90ec21041626c09f010b3ea1
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

1f2ce88a16878904240255bdf2514bd31051db0c7543ec26ef06de39357a3b09
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

253a68e1ffd47f4d88bdbdf54e8afc0197352e498876837a91033abddaa176b8
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

2fd6697a1f3b06d378b9af5a7c464c034937a2b92244cb020ad415eb3ebabfc2
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

33bf2aceb09b0b8914a23116071c866487285aaf2c07685eafe97e6bbbec84e6
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

3a0bb6eeb1acf731a04545412b27a045f2b7d82de7b2af8155c9b75ef45337dd
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

3fd6c2e13142bba3f6ecf03a9c82e361263f279442878ad462d998511c4bdfc1
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

4512fa5034fc4a78571e6b3b9713abbad990080bb50055b9af49988f15a9e323
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

47dce0b8f5af8a2c4065c47676e371323c8db08351f67f9a353da7739a0edc1d
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

4af4a5feed166e8ed1f64671310b4473a468723e79aec17e98572fdf8d07b6e5
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

5337d318cdad3c2f66142aa628ce738f7d406b02eacd81a6e83aa4509fc76c93
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

588e1c1ed5b08c2a16b97911077cda6051965f2956655724d38a41e2801033ad
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

61212693e38d8050637981330d47e0833c5ea01081f248dc2780e4db0421adee
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

623397bc9e69538234092ad089a6969c265e9223ae3b5d1bff76ca3d2ff0c848
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

64527906b47321de66553a25b4ba369d36fd00e6f326b1440c177fced1052455
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

6654a170a1307786509a27bbb404369ed9ced551ceff1396a190dbb0ecf48628
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

79a8ef16d11aff3ac3ecd40f144675dbc064d64717c4642fa19d4e23a3fc7bdd
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

7c334c61b941b8b2dd7da98f82f39657652fc3aee2c8d82ddaae277a47ed4c09
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

8047ff8e1a04ceb5b0070d5d741b4f6ff13abaa14d6ef8ab4e867a085f764dc1
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

8aebedc8a3b449b6fb89fae841f4f23a6a6e0e4d73a376350620f5232d1eb279
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

8b00f25abb68fb2624ce2af0df0aed7768a25ea1df07d1488a755533af063622
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

8ddb489fbc82776df984baf1ec0dc9f75b19374dbb2afda385a80eef2f3e4685
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

905addb68d5c0a7b6d7d3a63aebe8bde797e6cc46c4513371db1cdea92056ece
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

98d8daefa6c7efe226bebdcac85aec6d07787fc388d5577cc6b16b01ffb11730
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

9ae6e6c5ce4900bbf8692cdca732068a26da2a57f5bd1e9a881dd9af6e238794
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

9dbabfd1fade1e78ad5a34cb62f794f2247e739dacec1c517cc54cf882042df7
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

9eb16ea3e7032f210e7a12174641c73a6d4c5bf0694acde5ac04fc1efc990358
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

a288b6d463dbbffa6bf8a84cc9f6692dfc86b8ed19a70b98b784d6ee1c52f2c3
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

ab79d651e254325eebbb4a41ba415e6992d019a5ad40dbbf741b61cbf23c7f7c
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

b43ddef17b106924736e30c1611b85b500f4caf0d780cb24c29963c070e62146
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

b6719a2664f2cabb378cc3c8d5268e4aa66a3488c1586fc2867db1358980c20a
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

ba353277c252b373b83248d6ec158358ce1ec79fccd4cc269910ae018f4e724b
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

c63f1682413b0c1baba54a2da293ce6eebf0ed0c77a71c44137c792347cdafc7
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

c68e9be95a6a01e40d92127285f49e59abaa0fc97588bf4a9c0527ec399ee35f
    Windows Server Container,   Created,                                     , containerd-shim-runhcs-v1.exe

d831ce22dd0e7201488069b11298fc721a8f15c990b03dc213ae12dd4b752ace
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

dafcd3888adf1659b61cc0c5366d2b7af82f220c34c80dbfbd5956badf6e2f36
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

dd7e9fc29a063378013a81c28a86305cb2bd059f40efc7be3376e6fb957a301e
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

e04c298e170a49b652f90b996aecab6bdb3c6ee4826167bf7cdda9e7d6b0c9a3
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

ea9dab25e5f342345cd88e06ebe3195b27951d39d5ccb64abe19c60c41f63bd1
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

eb1ed76736f81aea0eff0112932a2b5a4f6fca84c207756ef60b14275658b1e8
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

eda5ddab54f07a587d217add77d07f20f338d596d3555a197dae4e20030c2439
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

f47ff9011ae9d9a99ddda4be706b7898c90feea01f8df5831fddc47b8274874c
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

f629e47e2660df6f7a46b26b35e0d3ceb237f29d4eee974fe3c26f710dd78bdc
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe

fb035618e2fb8471dd1a188ae3bc1226cbd35189480b5df2391d64582f8d00de
    Windows Server Container,   Running,                                     , containerd-shim-runhcs-v1.exe
PS C:\Users\capi\ProcessExplorer> get-date

Thursday, January 6, 2022 6:32:31 PM
```

This pod hasn't been removed until 6:35:

```
kubelet.exe.INFO:8852:I0106 18:35:03.876868    4788 generic.go:296] "Generic (PLEG): container finished"                podID=274e93ea-713e-4196-9a98-1a3c25a38f36                                                                              containerID="5e176e423efbb7dc745eeb980a8fe11731b0db5d8152b076c3feae3256dcdcbd" exitCode=128                             kubelet.exe.INFO:9030:I0106 18:36:31.949559    4788 fake_topology_manager.go:62] "RemoveContainer"                      containerID="5e176e423efbb7dc745eeb980a8fe11731b0db5d8152b076c3feae3256dcdcbd"
```


I couple graphs to demonstrate the load on the system:

![cpu spikes to 100% when running test and beyond](https://user-images.githubusercontent.com/648372/148436572-2857086f-3416-48e8-95c1-f872e39840e8.png)


![cpu drops when pods finally cleaned](https://user-images.githubusercontent.com/648372/148436584-768d34f2-e2f0-4eb0-97fc-d26e8cfdae12.png)



### Anything else we need to know?

This isn't really an issue on Linux, the container start/delete cycle is less intensive on the system.  Although there are a few containers still running after the test has completed they are cleaned up with in seconds after the test completes vs several minutes for windows.   

This mostly like because starting a container in windows isn't just running a single process, there are 8+ system process inside the container that need to be started, same for deletion.

### Relevant SIG(s)

/sig windows

/cc @claudiubelu @marosset

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind flake

/sig windows
/sig testing

#### What this PR does / why we need it:

Some of the garbage_collector tests spawn a large number of pods (50% of the current pod capacity). Running these tests
in parallel with other tests may cause them to fail because the kubelets will be too busy spawning these pods
to finish spawning the pods needed by other tests within the time limit (1 minute).

Spawning the pods in batches might offer other tests a chance to spawn their pods within their needed timeframe, reducing the
overall flakiness of test jobs.

#### Which issue(s) this PR fixes:

Fixes #107394

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by helayoty** [comment]
@claudiubelu  👋 the release 1.24 bug triage shadow. As we reached the code freeze phase, the PR will be removed from the release. Please submit an exception request if you feel like this has to land on 1.24

**Comment by helayoty** [comment]
@claudiubelu @deads2k  👋 the release 1.25 bug triage Lead. I'm reaching out to see the status of this PR and if we are still targeting the current release.

**Comment by claudiubelu** [comment]
> @claudiubelu @deads2k wave the release 1.25 bug triage Lead. I'm reaching out to see the status of this PR and if we are still targeting the current release.

Yes.

File: test/e2e/apimachinery/garbage_collector.go [hunk 1]
```diff
@@ -737,34 +759,19 @@ var _ = SIGDescribe("Garbage collector", func() {
...
-			if rc1.Status.Replicas == *rc1.Spec.Replicas {
-				return true, nil
-			}
-			return false, nil
-		}); err != nil {
-			framework.Failf("failed to wait for the rc.Status.Replicas to reach rc.Spec.Replicas: %v", err)
-		}
+
+		uniqLabelsDeleted := getUniqLabel("gctest_d", "valid_and_pending_owners_d")
+		rc1 := createOwnerRC(f, rc1Name, replicas, uniqLabelsDeleted)
```
Comment (lavalamp) [inline test/e2e/apimachinery/garbage_collector.go:773]: this changes the creation order, I wouldn't do that without a good reason

**Comment by lavalamp** [comment]
I'm not sure I endorse solving the problem this way, if users in a real cluster are interfering with each other the solution is hopefully to tune APF settings until the elephant gets throttled. Have we tried that?

**Pull request closed without merging**